### PR TITLE
feat: support `ExcalidrawElement.customData`

### PR DESCRIPTION
**Pull request by dwelle**
fix https://github.com/excalidraw/excalidraw/issues/5590

Currently it's not properly typed, just `Record<string, any>`. Later we may want to add support for typing it at least on a component-level (for all elements at once) — that will require more extensive refactor.

/cc @zsviczian I recall you needed this before?

### CODE REVIEW

**Review by ad1992 — COMMENTED** [review]
can you also update changelog and readme ?

File: src/packages/excalidraw/README.md [hunk 1]
```diff
@@ -470,6 +470,12 @@ This helps to load Excalidraw with `initialData`. It must be an object or a [pro
 
 You might want to use this when you want to load excalidraw with some initial elements and app state.
 
+#### Storing custom data on Excalidraw elements
+
+Beyond attributes that Excalidraw elements alredy support, you can store custom data on each element in a `customData` object. The type of the attribute is [`Record<string, any>`](https://github.com/excalidraw/excalidraw/blob/master/src/element/types.ts#L59) and is optional.
```
Comment (ad1992) [inline src/packages/excalidraw/README.md:475]:
```suggestion
Beyond attributes that Excalidraw elements already support, you can store custom data on each element in a `customData` object. The type of the attribute is [`Record<string, any>`](https://github.com/excalidraw/excalidraw/blob/master/src/element/types.ts#L59) and is optional.
```

File: src/packages/excalidraw/CHANGELOG.md [hunk 2]
```diff
@@ -17,7 +17,11 @@ Please add the latest change on the top under the correct section.
 
 #### Breaking Changes
 
-- `setToastMessage` API is now renamed to `setToast` API and the function signature is also updated [#5427](https://github.com/excalidraw/excalidraw/pull/5427). You can also pass `duration` and `closable` attributes along with `message`.
+- `setToastMessage` API is now renamed to `setToas○t` API and the function signature is also updated [#5427](https://github.com/excalidraw/excalidraw/pull/5427). You can also pass `duration` and `closable` attributes along with `message`.
+
+### Excalidraw Library
+
+- Added support for storing [`customData`](https://github.com/excalidraw/excalidraw/blob/master/src/packages/excalidraw/README.md#storing-custom-data-to-excalidraw-elements) on Excalidraw elements [#5592].
```
Comment (ad1992) [inline src/packages/excalidraw/CHANGELOG.md:24]: This should go under API instead as its for host (package)

File: src/packages/excalidraw/README.md [hunk 3]
```diff
@@ -470,6 +470,12 @@ This helps to load Excalidraw with `initialData`. It must be an object or a [pro
...
+
+You can use this to add any extra information you need to keep track of.
```
Comment (ad1992) [inline src/packages/excalidraw/README.md:477]: Also let's add that they can use `updateScene` /` updateLibrary` to add custom data once editor is loaded. I don't think its intutive right now that they could use these API's to update it

File: src/packages/excalidraw/CHANGELOG.md [hunk 4]
```diff
@@ -15,9 +15,11 @@ Please add the latest change on the top under the correct section.
 
 ### Excalidraw API
 
+- Added support for storing [`customData`](https://github.com/excalidraw/excalidraw/blob/master/src/packages/excalidraw/README.md#storing-custom-data-to-excalidraw-elements) on Excalidraw elements [#5592].
```
Comment (ad1992) [inline src/packages/excalidraw/CHANGELOG.md:18]:
```suggestion
#### Features

- Added support for storing [`customData`](https://github.com/excalidraw/excalidraw/blob/master/src/packages/excalidraw/README.md#storing-custom-data-to-excalidraw-elements) on Excalidraw elements [#5592].
```

File: src/packages/excalidraw/CHANGELOG.md [hunk 5]
```diff
@@ -15,9 +15,11 @@ Please add the latest change on the top under the correct section.
...
+
 #### Breaking Changes
 
-- `setToastMessage` API is now renamed to `setToast` API and the function signature is also updated [#5427](https://github.com/excalidraw/excalidraw/pull/5427). You can also pass `duration` and `closable` attributes along with `message`.
+- `setToastMessage` API is now renamed to `setToas○t` API and the function signature is also updated [#5427](https://github.com/excalidraw/excalidraw/pull/5427). You can also pass `duration` and `closable` attributes along with `message`.
```
Comment (ad1992) [inline src/packages/excalidraw/CHANGELOG.md:22]:
```suggestion
- `setToastMessage` API is now renamed to `setToast` API and the function signature is also updated [#5427](https://github.com/excalidraw/excalidraw/pull/5427). You can also pass `duration` and `closable` attributes along with `message`.
```

**Review by ad1992 — APPROVED** [review]
left few comments otherwise good to go 🚀

**Comment by zsviczian** [comment]
Just saw this one... looks super useful! I will see how I can convert my custom fields to this!

**Pull request merged**